openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml

```
@@ -1,51 +1,53 @@
+type: object
 description: |-
   Provides "buy X get Y" promotions.
   These promotions incentivize new customers to buy and also reward existing customers.
   This restriction type enables you to add discounted plans to an invoice when a certain quantity of other plans are purchased.
   For example, if a customer buys two or more books, they get one free bookmark.
-allOf:
-  - $ref: ../CouponRestriction.yaml
-  - $ref: ../RedemptionRestriction.yaml
-  - type: object
-    required:
-      - buy
-      - get
-    properties:
-      buy:
-        type: array
-        description: One or more plan and quantity definitions that must be purchased to receive all plans and quantities that are defined in the `get` instruction.
-        minItems: 1
-        items:
-          type: object
-          required:
-            - planId
-            - quantity
-          properties:
-            planId:
-              type: string
-              description: Plan ID of what is being purchased.
-              maxLength: 50
-              example: plan_0YV7DENSVGDBW9S71XZNNYYQ0X
-            quantity:
-              type: integer
-              minimum: 1
-              description: Total quantity of the associated plan ID that must be purchased for this restriction to apply.
-      get:
-        type: array
-        description: One or more plan and quantity definitions added to a purchase if all definitions in the `buy` condition are met.
-        minItems: 1
-        items:
-          type: object
-          required:
-            - planId
-            - quantity
-          properties:
-            planId:
-              type: string
-              description: Plan ID of what is to be received.
-              maxLength: 50
-              example: plan_0YV7DENSVGDBW9S71XZNNYYQ0X
-            quantity:
-              type: integer
-              minimum: 1
-              description: Total quantity of the associated plan ID that is received.
+required:
+  - type
+  - buy
+  - get
+properties:
+  type:
+    type: string
+    description: Coupon restriction type.
```

nitpick: 3 nouns in a row 
```suggestion
    description: Type of coupon restriction.
```